gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -1417,18 +1416,13 @@ bool GazeboRosApiPlugin::setModelState(gazebo_msgs::SetModelState::Request &req,
     if (relative_entity)
     {
       gazebo::math::Pose  frame_pose = relative_entity->GetWorldPose(); // - myBody->GetCoMPose();
-      gazebo::math::Vector3 frame_pos = frame_pose.pos;
-      gazebo::math::Quaternion frame_rot = frame_pose.rot;
 
-      //std::cout << " debug : " << relative_entity->GetName() << " : " << frame_pose << " : " << target_pose << std::endl;
-      //target_pose = frame_pose + target_pose; // seems buggy, use my own
-      target_pose.pos = model->GetWorldPose().pos + frame_rot.RotateVector(target_pos);
-      target_pose.rot = frame_rot * target_pose.rot;
+      target_pose = target_pose + frame_pose;
```

Comment:
From the docs of `operator+`:

```      /// \brief Addition operator
      /// A is the transform from O to P specified in frame O
      /// B is the transform from P to Q specified in frame P
      /// then, B + A is the transform from O to Q specified in
 frame O
```

So shouldn't this be the other way around as previously proposed in the commented code?

Reply:
Please correct me if I'm wrong. The way I see it is we want to specify the target pose in a different frame, so the target pose equates to the transform from P to Q, while the frame is then the transform O to P?

Reply:
You are right, I'm sorry. I was too used to a different conversion.
I just tested your code and it works well.